tests/metropolis_hasting_producer_12.cc

```diff
@@ -0,0 +1,63 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2020 by the SampleFlow authors.
+//
+// This file is part of the SampleFlow library.
+//
+// The SampleFlow library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of SampleFlow.
+//
+// ---------------------------------------------------------------------
+
+
+// Test the Metropolis-Hastings producer with a diagonally elongated
+// distribution.
+
+#include <iostream>
+#include <sampleflow/producers/metropolis_hastings.h>
+#include <sampleflow/filters/conversion.h>
+#include <sampleflow/consumers/mean_value.h>
+#include <valarray>
+#include <random>
+#include <cmath>
+
+
+using SampleType = std::valarray<double>;
+
+
+double log_likelihood (const SampleType &x)
+{
+  std::valarray<double> mu = {0, 0};
+  std::valarray<double> cov = {{1, 1.5}, {1.5, 3}};
+  // FIXME: what's the correct way to do these matrix operations?
+  return -0.5 * (ln(|cov|) + (x - mu)^T @ cov^-1 @ (x - mu) + k * ln(2 * pi));
```

Comment:
First, the `log_likelihood` doesn't have to be normalized, so you can remove the `ln(|cov|)` factor along with the `ln(2*pi)` one.

Second, just multiply it out:
```suggestion
  return -0.5 * ( (x[0]-mu[0])*cov[0][0]*(x[0]-mu[0]) +
                        (x[0]-mu[0])*cov[0][1]*(x[1]-mu[1]) +
                        (x[1]-mu[1])*cov[1][0]*(x[0]-mu[0]) +
                        (x[1]-mu[1])*cov[1][1]*(x[1]-mu[1]) +
```
That's not elegant, but it'll do for this test.